When an error occurs in the Camel MQTT endpoint, the CallbackConnection onFailure callback simply disconnects the connection.
At that point there doesn't seem to be any mechanism to reconnect to the broker.

            public void onFailure(Throwable value) {

                connection.disconnect(new Callback<Void>() {

                    public void onSuccess(Void value) {

                    }



                    public void onFailure(Throwable e) {

                        LOG.debug("Failed to disconnect from " + configuration.getHost() + ". This exception is ignored.", e);

                    }

                });

            }


